project/src/test/java/com/google/sps/ServletExampleWithInjectionTest.java

```diff
@@ -0,0 +1,40 @@
+package com.google.sps;
+
+import static org.junit.Assert.*;
+
+import com.google.sps.data.User;
+import com.google.sps.servlets.ServletExampleWithInjection;
+import com.google.sps.utilities.DatabaseWrapper;
+import com.google.sps.utilities.TestDatabaseWrapper;
+import java.io.*;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.*;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(JUnit4.class)
+public class ServletExampleWithInjectionTest extends Mockito {
+  private final DatabaseWrapper wrapper = new TestDatabaseWrapper();
+  private final ServletExampleWithInjection servlet = new ServletExampleWithInjection();
+
+  @Before
+  public void setUp() throws Exception {
+    servlet.setWrapper(wrapper);
```

Comment:
The thought is correct, but I don't think you particularly use injection here.

I think in the example I gave you, it uses a Module (PetTestModule) to config which Pet to use. So you don't need to call setPet function, the Pet instance is injected to the class.

Could you maybe try to use a Module to control which Wrapper to use? Let me know if you have trouble to find how to do so.

Reply:
Thanks, so my thinking was that since this is a Servlet, it is not recommended to use the constructor to inject a parameter (I saw some resources like this https://stackoverflow.com/questions/25868944/servlet-does-not-have-a-default-constructor). That is why I opted to use a set function.

Let me know if you think I can still try to use a module to control which wrapper to use.

Reply:
I found some questions on Injection in Servlet, could you take a look and let me know if any of them works?

https://www.google.com/search?q=servlet+injection+site:stackoverflow.com&sxsrf=ALeKk02c8uEN_mQb3c4rRUvBOykLjtMEtw:1596217023377&sa=X&ved=2ahUKEwjU96Llg_jqAhXPvZ4KHZmHB5MQrQIoBHoECAYQBQ&biw=1120&bih=611&dpr=3

Reply:
Will do, thanks.